### PR TITLE
[Parser][SE-0034] Replace line directive #line with #setline

### PR DESCRIPTION
**Pull request by dduan**
#### What's in this pull request?

Replace line directive `#line` with `#setline`. This change was proposed in [SE-0034](https://github.com/apple/swift-evolution/blob/master/proposals/0034-disambiguating-line.md).
#### Resolved bug number: ([SR-840](https://bugs.swift.org/browse/SR-840))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by lattner** [comment]
@swift-ci please test

File: lib/Parse/ParseStmt.cpp [hunk 1]
```diff
@@ -291,7 +292,7 @@ ParserStatus Parser::parseBraceItems(SmallVectorImpl<ASTNode> &Entries,
     PreviousHadSemi = false;
     if (isStartOfDecl()
         && Tok.isNot(tok::pound_if)
-        && Tok.isNot(tok::pound_line)) {
+        && Tok.isNot(tok::pound_setline)) {
```
Comment (lattner) [inline lib/Parse/ParseStmt.cpp]: Minor nitpick, you can use Tok.isNot(x, y) here, though it looks like a preexisting issue.

**Comment by lattner** [comment]
Overall, the patch looks great to me, thanks!  Random other question: does gyb or any other tool produce these markers?  If so, it should be updated to use the new syntax.

**Comment by dduan** [comment]
@lattner Thanks for reminding me of gyb. There are a couple of default argument and documentation set to `#line`. I updated them and am running tests locally now.

**Comment by dduan** [comment]
gyb updated. Tests are passing locally. Can't wait to be able to trigger CI testing myself :)

**Comment by lattner** [comment]
Awesome, nice job, please commit!

**Pull request merged**